metadata-ingestion/docs/sources/kafka-connect/README.md

```diff
@@ -16,9 +36,138 @@ This ingestion source maps the following Source System Concepts to DataHub Conce
+| **S3 Sink**<br/>`io.confluent.connect.s3.S3SinkConnector`                      | ✅ Full             | ✅ Full                 | Runtime API / Config-based | Topic → S3 object mapping |
+| **Snowflake Sink**<br/>`com.snowflake.kafka.connector.SnowflakeSinkConnector`  | ✅ Full             | ✅ Full                 | Runtime API / Config-based | Topic → Table mapping     |
+| **Cloud PostgreSQL Sink**<br/>`PostgresSink`                                   | ✅ Full             | ✅ Full                 | Runtime API / Config-based | Topic → Table mapping     |
+| **Cloud MySQL Sink**<br/>`MySqlSink`                                           | ✅ Full             | ✅ Full                 | Runtime API / Config-based | Topic → Table mapping     |
+| **Cloud Snowflake Sink**<br/>`SnowflakeSink`                                   | ✅ Full             | ✅ Full                 | Runtime API / Config-based | Topic → Table mapping     |
+
+**Legend:**
+
+- ✅ **Full**: Complete lineage extraction with accurate topic discovery
+- ✅ **Partial**: Lineage extraction supported but topic discovery may be limited (config-based only)
+- ❌ **Manual**: Requires `generic_connectors` configuration for lineage mapping
+
+### Supported Transforms
+
+DataHub uses an **advanced transform pipeline strategy** that automatically handles complex transform chains by applying the complete pipeline to all topics and checking if results exist. This provides robust support for any combination of transforms.
+
+#### Topic Routing Transforms
+
+- **RegexRouter**: `org.apache.kafka.connect.transforms.RegexRouter`
+- **Cloud RegexRouter**: `io.confluent.connect.cloud.transforms.TopicRegexRouter`
+- **Debezium EventRouter**: `io.debezium.transforms.outbox.EventRouter` (Outbox pattern)
+
+#### Non-Topic Routing Transforms
+
+DataHub recognizes but passes through these transforms (they don't affect lineage):
+
+- InsertField, ReplaceField, MaskField, ValueToKey, HoistField, ExtractField
+- SetSchemaMetadata, Flatten, Cast, HeadersFrom, TimestampConverter
+- Filter, InsertHeader, DropHeaders, Drop, TombstoneHandler
+
+#### Transform Pipeline Strategy
+
+DataHub uses an improved **reverse transform pipeline approach** that:
+
+1. **Takes all actual topics** from the connector manifest/Kafka cluster
+2. **Applies the complete transform pipeline** to each topic
+3. **Checks if transformed results exist** in the actual topic list
+4. **Creates lineage mappings** only for successful matches
+
+**Benefits:**
+
+- ✅ **Works with any transform combination** (single or chained transforms)
+- ✅ **Handles complex scenarios** like EventRouter + RegexRouter chains
+- ✅ **Uses actual topics as source of truth** (no prediction needed)
+- ✅ **Future-proof** for new transform types
+- ✅ **Works identically** for both self-hosted and Confluent Cloud environments
```

Comment:
this is relevant for lineage, right?

as mentioned in another comment, we should explain why this is relevant to the user